x-pack/plugin/core/src/main/resources/ecs-template.yml

```diff
@@ -0,0 +1,225 @@
+---
+template:
+  settings:
+    # This is a made up setting which does not exist today. The idea is that
+    # any field that shows up after the first 1000 fields is kept in source
+    # but does not show up as runtime for index field
+    #index.mapping.total_fields.soft_limit: 1000
+
+    # Ignores any malformed data. Data will still be persisted in _source
+    index.mapping.ignore_malformed": true
+
+  mappings:
+
+    # Any field that is discovered is mapped as a runtime field
+    dynamic: runtime
+
+    # DISCUSS: What are the pros / cons of having this enabled?
+    date_detection: false
+
+    ### All the properties which are set in each template
+    properties:
+
+      # This ensures all subobjects are flattened and prevents and object / keyword conflicts
+      # As subobject is set to false, no objects like host, error or similar are mapped.
+      subobject: false
+
+      # @timestamp is the only field that MUST exist.
+      "@timestamp":
+        type: date
+        ignore_malformed: false
+
+      # The data stream properties are required in any index following the data stream naming scheme
+      # Discuss: If there is an ecs: enabled flag it can also be used in indices so this would have to be skipped
+      # These fields might not even be needed in this template
+      data_stream.type:
+        type: constant_keyword
+        #value: logs
+      data_stream.dataset:
+        type: constant_keyword
+      data_stream.namespace:
+        type: constant_keyword
+
+
+    # To ensure no empty mappings are in an index, everything except the properties above are mapped as dynamic templates
+    # The dynamic mappings here are a best effort to map ECS core fields: https://github.com/elastic/ecs/blob/8.4/generated/csv/fields.csv
+    # As most fields are keyword, these will be automatically be matched correctly. Only core fields were taken into account.
+    # DISCUSS: Should @timestamp and data_stream.* also be moved under dynamic templates?
+    dynamic_templates:
+
+      ### Index mappings
+
+      # Message field is always index
+      - message:
+          path_match: message
+          mapping:
+            type: match_only_text
+
+      ### Runtime mappings
+
+      # Matching all string fields by default to keyword
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+            # Do we need this for runtime?
+            ignore_above: 1024
+
+      # Matching all long fields
+      # This matches the following fields: bytes, packets, port, duration, severity, pid
+      # No path matching is needed as it is assumed these fields are already in the json doc a long
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
```

Comment:
The part that is important for me here is line 70 which documents which ECS fields it should match.

Reply:
BTW maybe a middle ground that could be found is leave the comments in but comment out the full block for documentation purpose. My assumption is that as soon as this would get loaded into ES / cluster state, comments would be kicked out anyways.